src/main/java/de/siphalor/mousewheelie/client/inventory/ContainerScreenHelper.java

```diff
@@ -12,121 +12,132 @@
+                    if (slot.getStack().isItemEqualIgnoreDamage(referenceStack)) {
+                        if (slot.getStack().getCount() < stackSize) {
+                            stackSize = slot.getStack().getCount();
+                            moveSlot = slot;
+                            if (stackSize == 1) break;
+                        }
+                    }
+                }
+                if (moveSlot != null)
+                    sendSingleItem(moveSlot);
+            }
+        }
+    }
 
-	public boolean shallChangeInventory(Slot slot, boolean scrollUp) {
-		return isLowerSlot(slot) == scrollUp;
-	}
+    public boolean shallChangeInventory(Slot slot, boolean scrollUp) {
+        return isLowerSlot(slot) == scrollUp;
+    }
 
-	public boolean isHotbarSlot(Slot slot) {
-		return ((ISlot) slot).mouseWheelie_getInvSlot() < 9;
-	}
+    public boolean isHotbarSlot(Slot slot) {
+        return ((ISlot) slot).mouseWheelie_getInvSlot() < 9;
+    }
 
-	public boolean isLowerSlot(Slot slot) {
-		if (screen instanceof AbstractInventoryScreen) {
-			return isHotbarSlot(slot);
-		} else {
-			return (slot.inventory instanceof PlayerInventory);
-		}
-	}
+    public boolean isLowerSlot(Slot slot) {
+        if (screen instanceof AbstractInventoryScreen) {
+            return isHotbarSlot(slot);
+        } else {
+            return (slot.inventory instanceof PlayerInventory);
+        }
+    }
 
-	public void sendSingleItem(Slot slot) {
-		clickHandler.handleClick(slot, 0, SlotActionType.PICKUP);
-		clickHandler.handleClick(slot, 1, SlotActionType.PICKUP);
-		clickHandler.handleClick(slot, 0, SlotActionType.QUICK_MOVE);
-		clickHandler.handleClick(slot, 0, SlotActionType.PICKUP);
-	}
+    public void sendSingleItem(Slot slot) {
+        clickHandler.handleClick(slot, 0, SlotActionType.PICKUP);
+        clickHandler.handleClick(slot, 1, SlotActionType.PICKUP);
+        clickHandler.handleClick(slot, 0, SlotActionType.QUICK_MOVE);
+        clickHandler.handleClick(slot, 0, SlotActionType.PICKUP);
+    }
 
-	public void sendStack(Slot slot) {
-		clickHandler.handleClick(slot, 0, SlotActionType.QUICK_MOVE);
-	}
+    public void sendStack(Slot slot) {
+        clickHandler.handleClick(slot, 0, SlotActionType.QUICK_MOVE);
+    }
 
-	public void sendAllOfAKind(Slot referenceSlot) {
-		ItemStack referenceStack = referenceSlot.getStack().copy();
-		for (Slot slot : screen.getContainer().slots) {
-			if (slotsInSameScope(slot, referenceSlot)) {
-				if (slot.getStack().isItemEqualIgnoreDamage(referenceStack))
-					clickHandler.handleClick(slot, 0, SlotActionType.QUICK_MOVE);
-			}
-		}
-	}
+    public void sendAllOfAKind(Slot referenceSlot) {
+        ItemStack referenceStack = referenceSlot.getStack().copy();
+        for (Slot slot : screen.getContainer().slots) {
+            if (slotsInSameScope(slot, referenceSlot)) {
+                if (slot.getStack().isItemEqualIgnoreDamage(referenceStack)) {
+                    if (hasAltDown()) {
```

Comment:
The `send*` methods are meant to be used as helper methods only for directly transferring items to another inventory.
Since scrolling behavior relies on these methods they shouldn't check for modifier keys on their own.

Instead there the modifier check should be performed in the mixin and there should be separate methods for dropping stacks (`dropAllOfAKind`, `dropAllFrom` etc).

Reply:
it fixed in new commit